src/services/submissionService.ts

```diff
@@ -377,6 +386,142 @@ function validateIdInput(id: string): Result<unknown> {
+    !validOrderByFields.includes(order.fieldToOrder))
+
+if (invalidOrderByOption) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'orderBy',
+        errorCode: ErrorCode.ORDERBY_OPTION_NOT_FOUND,
+        httpStatus: 400
+    })
+}
+
+if (filters.orderBy && filters.orderBy.length > 2) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'orderBy',
+        errorCode: ErrorCode.ORDERBY_OPTION_MAX_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.orderBy && filters.orderBy.length === 2 && 
+    filters.orderBy[0].fieldToOrder === filters.orderBy[1].fieldToOrder) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'orderBy',
+        errorCode: ErrorCode.ORDERBY_FIELD_SELECTED_TWICE,
+        httpStatus: 400
+    })
+}
+
+if (filters.limit && filters.limit > 1000) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'limit',
+        errorCode: ErrorCode.MAX_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.limit && filters.limit < 0) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'limit',
+        errorCode: ErrorCode.MIN_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.offset && filters.offset > 10000) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'offset',
+        errorCode: ErrorCode.MAX_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.offset && filters.offset < 0) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'offset',
+        errorCode: ErrorCode.MIN_LIMIT,
+        httpStatus: 400
+    })
+}
+
+if (filters.googleMapsUrl === '' || (filters.googleMapsUrl && !filters.googleMapsUrl.trim())) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'googleMapsUrl',
+        errorCode: ErrorCode.MISSING_INPUT,
+        httpStatus: 400
+    })
+}
+
+if (filters.healthcareProfessionalName === '' || (filters.healthcareProfessionalName && !filters.healthcareProfessionalName.trim())) {
+    validateSearchResults.hasErrors = true
+    validateSearchResults.errors?.push({
+        field: 'healthcareProfessionalName',
+        errorCode: ErrorCode.MISSING_INPUT,
+        httpStatus: 400
+    })
+}
+
+if (filters.spokenLanguages) {
+    if (filters.spokenLanguages.length === 0) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'spokenLanguages',
+            errorCode: ErrorCode.MISSING_INPUT,
+            httpStatus: 400
+        })
+    } else {
+        for (const lang of filters.spokenLanguages) {
+            if (!Object.values(gqlTypes.Locale).includes(lang)) {
+                validateSearchResults.hasErrors = true
```

Comment:
I have added a check for a valid language code, but it seems the `Locale` enum already does this, so maybe this isn’t needed?